File: testdata/jennies/rawtypes/dashboard/PHPRawTypes/src/Dashboard/FieldConfigSource.php
<?php

namespace Grafana\Foundation\Dashboard;

class FieldConfigSource implements \JsonSerializable
{
    public ?\Grafana\Foundation\Dashboard\FieldConfig $defaults;

    /**
     * @param \Grafana\Foundation\Dashboard\FieldConfig|null $defaults
     */
    public function __construct(?\Grafana\Foundation\Dashboard\FieldConfig $defaults = null)
    {
        $this->defaults = $defaults;
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if (($this->defaults === null) !== ($other->defaults === null)) {
            return false;
        }
        if ($this->defaults !== null) {
            if (!$this->defaults->equals($other->defaults)) {
                return false;
            }
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{defaults?: mixed} $inputData */
        $data = $inputData;
        return new self(
            defaults: isset($data["defaults"]) ? (function($input) {
    	/** @var array{unit?: string, custom?: mixed} */
    $val = $input;
    	return \Grafana\Foundation\Dashboard\FieldConfig::fromArray($val);
    })($data["defaults"]) : null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        if (isset($this->defaults)) {
            $data->defaults = $this->defaults;
        }
        return $data;
    }
}

File: testdata/jennies/rawtypes/dashboard/JavaRawTypes/dashboard/FieldConfigSource.java
package dashboard;

import java.util.Objects;

public class FieldConfigSource {
    public FieldConfig defaults;
    public FieldConfigSource() {
    }
    public FieldConfigSource(FieldConfig defaults) {
        this.defaults = defaults;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof FieldConfigSource)) return false;
        FieldConfigSource o = (FieldConfigSource) other;
        if (!Objects.equals(this.defaults, o.defaults)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.defaults);
    }
}
